### PR TITLE
Upgrade schemapack to 2.0.0-alpha3 in 'schemapack' branch

### PR DESCRIPTION
**Pull request by lkuchenb**
This patch migrates the `schemapack` branch to schemapack 2.0.0-a3. The previously migrated transformations have been migrated to the current schemapack 2.x standard.

## Notable changes

* The multiplicity parameter of the infer relations transformation has been removed. I believe the multiplicity and mandatoriness :-) can be inferred as implemented [here](https://github.com/ghga-de/metldata/blob/1110a3378f301985e1a699ebab6c13a0fb6d3c72/src/metldata/builtin_transformations/infer_relations/model_transform.py#L38-L118). However, maybe these should go to the `path` submodule (which in turn should move up the module hierarchy if we are going to re-use it in other transformations as planned)?
* I've added a placeholder `dev_launcher` script to re-enable Dockerfile updates through the template update mechanism.

### CODE REVIEW

File: src/metldata/builtin_transformations/infer_relations/model_transform.py [hunk 1]
```diff
@@ -17,19 +17,107 @@
...
+    """Infer the multiplicity of an inferred relation based on the path.
+
+    Args:
+        path: The path to infer the multiplicity for.
+        schema: The underlying schema.
+
+    Returns:
+        The inferred multiplicity.
+    """
+    origin, target = False, False
```
Comment (Cito) [inline src/metldata/builtin_transformations/infer_relations/model_transform.py:67]:
```suggestion
    origin = target = False
```

File: src/metldata/builtin_transformations/infer_relations/model_transform.py [hunk 2]
```diff
@@ -17,19 +17,107 @@
...
+    """Infer the mandatory property of an inferred relation based on the path.
+
+    Args:
+        path: The path to infer the mandatory property for.
+        schema: The underlying schema.
+
+    Returns:
+        The inferred mandatory property.
+    """
+    origin, target = True, True
```
Comment (Cito) [inline src/metldata/builtin_transformations/infer_relations/model_transform.py:100]:
```suggestion
    origin = target = True
```

File: src/metldata/builtin_transformations/infer_relations/model_transform.py [hunk 3]
```diff
@@ -17,19 +17,107 @@
...
+        schema: The underlying schema.
+
+    Returns:
+        The relation object.
+    """
+    if element.type_ == RelationPathElementType.ACTIVE:
+        class_ = schema.classes[element.source]
+    else:
+        class_ = schema.classes[element.target]
+    return class_.relations[element.property]
```
Comment (Cito) [inline src/metldata/builtin_transformations/infer_relations/model_transform.py:52]:
```suggestion
    as_source = element.type_ == RelationPathElementType.ACTIVE
    class_name = element.source if as_source else element.target
    return schema.classes[class_name].relations[element.property]
```

File: src/metldata/builtin_transformations/infer_relations/model_transform.py [hunk 4]
```diff
@@ -72,6 +161,6 @@ def add_inferred_relations(
             }
         )
 
-    return model.model_copy(
-        update={"classes": FrozenDict({**model.classes, **updated_class_defs})}
-    )
+    model_dict = model.model_dump()
+    model_dict.update({"classes": {**model.classes, **updated_class_defs}})
```
Comment (Cito) [inline src/metldata/builtin_transformations/infer_relations/model_transform.py:165]:
```suggestion
    model_dict['classes'].update(updated_class_defs)
```

**Review by Cito — APPROVED** [review]
Looks good to me.

When running the tests, I get a deprecation warning from pydantic and another user warning. These should be fixed in another PR.

**Comment by lkuchenb** [comment]
Thanks @Cito, ready for another round

**Review by Cito — APPROVED** [review]

**Pull request merged**